[AMDGPU] Use a generic printer for NamedIntOperands.

kosarev opened the pull request:
This includes simplifying printing dmask modifiers where we don't need to mask the value to print.

Part of <https://github.com/llvm/llvm-project/issues/62629>.

jayfoad commented on llvm/lib/Target/AMDGPU/MCTargetDesc/AMDGPUInstPrinter.cpp:
```diff
@@ -1812,14 +1711,18 @@ void AMDGPUInstPrinter::printEndpgm(const MCInst *MI, unsigned OpNo,
+  int64_t V = MI->getOperand(OpNo).getImm();
+  if (AlwaysPrint || V != 0) {
+    if (Width)
+      V &= maxUIntN(Width);
```
This should go before the `V != 0` check, otherwise you can end up printing `0` when `AlwaysPrint` is false.

kosarev replied:
I think if we drop significant bits here, then this should rather suggest the operand has a value that it's not supposed to have. This made me wondering why do we need the masking in the first place, which turns out we don't.

kosarev committed:
[AMDGPU][NFCI] Use a generic printer for NamedIntOperands.

This includes simplifying printing dmask modifiers where we
don't need to mask the value to print.

Part of <https://github.com/llvm/llvm-project/issues/62629>.

arsenm reviewed (approved):
We don't really have the proper validation that operands don't have out of range values
